plugins/okta/pkg/okta/okta.go

```diff
@@ -227,6 +235,23 @@ func (oktaPlugin *Plugin) Extract(req sdk.ExtractRequest, evt sdk.EventReader) e
 
 		oktaPlugin.lastLogEvent = data
 		oktaPlugin.lastEventNum = evt.EventNum()
+
+		if data.EventType == "user.mfa.okta_verify.deny_push" || (data.EventType == "user.authentication.auth_via_mfa" && data.Outcome.Result == "FAILURE") {
+			key := data.EventType + ":" + data.Actor.ID
+			now := time.Now().UnixNano()
+			valueList := []string{}
+			value, err := oktaPlugin.cache.Get(key)
+			if err == nil {
+				if value != nil {
+					valueList = strings.Split(value.(string), ",")
+				}
+			}
+			valueList = append(valueList, fmt.Sprintf("%v", now))
+			err = oktaPlugin.cache.SetWithExpire(key, strings.Join(valueList, ","), time.Duration(oktaPlugin.CacheExpiration)*time.Second)
```

Comment:
I had same idea, but I didn't know how to dynamically find the value. If you know, I'll be happy to change. I'll just add few seconds the expiration by security

Reply:
The same method can be used to cap the value of `valueList` (its max length should always be `oktaPlugin.maxShift`) at runtime.

Also, here we're using `req.ArgKey()` all over the place, but since we have a positive integer parameter I would declare the field arg with `IsIndex: true`, so that `req.ArgIndex()` can do all the parsing for us out of the box (and the framework will return formatting errors automatically).

Reply:
We can't have a first round with 0 as expiration, the key disappears immediately. I wasn't able to find a nice way to dynamically set up the TTL for keys, so I kept the parameter to configure `cache_expiration`, with a default value to 24h.
The global cache is capped to 10000 keys (hardcoded, I think it's enough and even too big), and now, with the `cache_usermaxsize` we cap the length of each keys.
We need to keep in mind the goal is to store the failed or denied MFA requests, it's useless to store too much of them and for too long, because the idea is to trigger an event when too much occurrences happen in a short period of time, as it might be related to an attempt to hack.